packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManagerMapBuffer.java

```diff
@@ -419,6 +419,7 @@ public static void adjustSpannableFontToFit(
     int initialFontSize = currentFontSize;
     while (currentFontSize > minimumFontSize
         && ((maximumNumberOfLines != ReactConstants.UNSET
+               && maximumNumberOfLines != 0
```

Comment:
I've added a second check, instead of replacing it to match https://github.com/facebook/react-native/blob/9e9fbec475ff99e9e86d63e34bd39905c30e85e3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManagerMapBuffer.java#L521